### ISSUE TITLE
Nested blocks and functions get weird visual indent

### ISSUE THREAD

**Issue by xfix** [rust-lang-nursery/rustfmt#1570]
I honestly don't know how to describe that one. It's like rustfmt cannot decide whether it wants to include alignment with a block to indent. Default settings (like on play.rust-lang.org).

This:

```rust
fn example() {
    a_very_long_function_name({some_func(1, {1})})
}
```

Changes into this:

```rust
fn example() {
    a_very_long_function_name({
                                  some_func(1, {
            1
        })
                              })
}
```

**Comment by zonyitoo** [comment]
I have encountered the same problem, which messes up the whole file!!

**Comment by nrc** [comment]
This looks pretty bad with the RFC formatting too:

```
fn example() {
    a_very_long_function_name(
        {
            some_func(
                1, {
                    1
                }
            )
        }
    )
}
```

There should be some elision of line breaks, but once we multi-line the second arg of the nested function, it should put the args on different lines.

**Comment by topecongiro** [comment]
Which is expected when using RFC? 
```rust
fn example() {
    a_very_long_function_name({some_func(1, {
        1
    })})
}
```
```rust
fn example() {
    a_very_long_function_name({
        some_func(1, {
            1
        })
    })
}
```
```rust
fn example() {
    a_very_long_function_name({
        some_func(
            1,
            {
                1
            },
        )
    })
}
```

**Comment by nrc** [comment]
I would expect both the blocks to be single-line because they are single-expression blocks, so I'd probably expect:

```
fn example() {
    a_very_long_function_name({ some_func(1, { 1 }) })
}
```

If the inner block were not, then I would expect your first example (because everything nests and in the two arg case, we keep the last arg on the same line as the prefix args). However, I would expect a space after the opening `{`. I'd expect the second example if the outer block had a semi-colon too. I think you'd only get the third example if the first `1` were actually very long.

**Comment by topecongiro** [comment]
@nrc Thank you for clarifying!

**Comment by topecongiro** [comment]
Closed via #1729.
